cmd/server/cadence/server.go

```diff
@@ -175,7 +175,7 @@ func (s *server) startService() common.Daemon {
 	)
 
 	if s.cfg.PublicClient.HostPort != "" {
```

Comment:
Minor suggestion/nit
```suggestion
	if len(s.cfg.PublicClient.HostPort) == 0 {
		log.Fatalf("need to provide an endpoint config for PublicClient")
        }
        
      params.DispatcherProvider = rpc.NewDispatcherProvider(params.Logger, rpc.NewDNSPeerChooserFactory(s.cfg.PublicClient.RefreshInterval, params.Logger))
```